### PR TITLE
Complete android user tracking SDK

### PR DESCRIPTION
**Pull request by ChenXiaoman**
The codes are all implemented. However, the SDK is not created yet

### CODE REVIEW

File: usertrackingsdk/src/main/java/com/visenze/usertrackingsdk/TrackingParams.java [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+     *               e.g. "add-to-wish-list", "view", "click"
+     * @param requestId The request id of the current search provided
+     *                  by ViSenze search service
+     */
+    public TrackingParams(String action, String requestId) {
+        this.action = action;
+        this.requestId = requestId;
+        customParams = new HashMap<String, String>();
+    }
+
```
Comment (yulu) [inline usertrackingsdk/src/main/java/com/visenze/usertrackingsdk/TrackingParams.java:30]: I think we can add another constructor which accepts action, requestId and customParams as parameters

File: usertrackingsdk/src/main/java/com/visenze/usertrackingsdk/UserEventTracker.java [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+     *            e.g. app key
+     */
+    public UserEventTracker(Context context, String accessKey) {
+        this.accessKey = accessKey;
+        httpInstance = HttpInstance.getInstance(context.getApplicationContext());
+        UIDManager.initUIDManager(context);
+    }
+
+    public void track(TrackingParams trackParams) {
+        httpInstance.addGetRequestToQueueWithoutResponse("http://track.visenze.com/__aq.gif", trackParams.toMap(accessKey));
```
Comment (yulu) [inline usertrackingsdk/src/main/java/com/visenze/usertrackingsdk/UserEventTracker.java]: make this endpoint "http://track.visenze.com/__aq.gif" configurable, default as this one

**Review by yulu — COMMENTED** [review]
please fix the codeclimate issues

File: usertrackingsdk/src/main/java/com/visenze/usertrackingsdk/TrackingParams.java (hunk 1 above)
Comment (ChenXiaoman) [reply usertrackingsdk/src/main/java/com/visenze/usertrackingsdk/TrackingParams.java:30]: Added

File: usertrackingsdk/src/main/java/com/visenze/usertrackingsdk/UserEventTracker.java (hunk 2 above)
Comment (ChenXiaoman) [reply usertrackingsdk/src/main/java/com/visenze/usertrackingsdk/UserEventTracker.java]: Made

**Pull request closed without merging**